### DISCUSSION TITLE
How can I set custom header on data fetching (useLoaderData)

### DISCUSSION THREAD

**Discussion by sunnhyn** [remix-run/remix#4638] [Q&A] [answered]
Hi, I'm trying to make some authorization logic on remix.
In my authorization logic, store JWT refresh token on https-only cookie and store access token at local variables (maybe recoil if possible). And then, I want to send request (both From request and just fetching data) with my access token on header.
I'm trying to do this, but i can't see 'authorization' field at header on loader function.

```typescript
export const headers: HeadersFunction = () => ({
  "authorization": "My JWT access token will be here",
});

export const loader: LoaderFunction = async ({ request }) => {
  console.log({ headers: request.headers });
  // ..do something with authorization header
}

export default function Index() {
  const data = useLoaderData();
  // ...
}
```
I found #4397, but it is about form request, not about loader function.
And.. I can not find 'header' props at  \<Form /\>'s prop typing (FormProps).

I'm a beginner in Remix, so I think I'll be able to learn thankfully if I missed something.
thanks for your help. 👍

**Comment by sergiodxa** [comment, marked as answer]
There's no way to attach headers to forms in HTML, so there's also no way to do it with Remix's Form component because it uses the same API.

If you store your JWT in a cookie, you don't need the Authorization header, you can get it from `request.headers.get("cookie")`, there you will receive all cookies the browser has stored for your domain, you can pass that string you receive to a [Cookie container](https://remix.run/docs/en/v1/api/remix#cookies) or [SessionStorage object](https://remix.run/docs/en/v1/api/remix#sessions), depending on what you used to save the JWT in the cookies.

Also, the HeadersFunction is to send headers from the server to the browser when the response is a document response (it send HTML and not JSON), not to receive headers on your loaders or actions.

**Reply by sunnhyn** [reply]
I understand. Thank you for answer and great framework.
Have a nice day 😃
